### PR TITLE
Nw4- JavaScript-Core-1 -Course-Week2- Bimbola Lasisi

### PR DESCRIPTION
**Pull request by blasisi**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Review by humailhasankhan — COMMENTED** [review]
Nicely done Bimbola. I seem to find a lot of similarity in our approaches. Your code is very clear and I would have probably done it the same way. I like that you think out of the box and you changed daniel's role and used not true to find your answers. Just a few things I would have done differently.  For the radio stations exercise, since we were meant to use a function to create an array, I used a for loop (let i = 87; i <= 108; i++) and then used .push inside an empty array I declared before so the frequencies are generated automatically, through the function. I am sure it would have still passed the required tests. I think you missed two functions in Exercise D: isShortName and startsWithD. Otherwise, great job. I really enjoyed going through your code and it is very clear and readable. Nicely done once again!.

File: exercises/C-comparison-operators/exercise.js [hunk 1]
```diff
@@ -7,14 +7,14 @@
 
 var studentCount = 16;
 var mentorCount = 9;
-var moreStudentsThanMentors; // finish this statement
+var moreStudentsThanMentors = studentCount > mentorCount; // finish this statement
 
 var roomMaxCapacity = 25;
-var enoughSpaceInRoom; // finish this statement
+var enoughSpaceInRoom = roomMaxCapacity != enoughSpaceInRoom; // finish this statement
```
Comment (seyedsharifi12) [inline exercises/C-comparison-operators/exercise.js:14]: Hi Bimbola, In here you need to show if there is enough space in room for students and mentors, in other words you need to assign a value to enoughSpaceInRoom variable that shows if you have enough room for all the students and mentors. for example if max capacity of the room was more than or equal to the combined number of students and mentors then you will have enough room.

File: exercises/C-comparison-operators/exercise.js [hunk 2]
```diff
@@ -7,14 +7,14 @@
...
 
 var personA = "Daniel";
 var personB = "Irina";
-var sameName; // finish this statement
+var sameName = personA >= personB; // finish this statement
```
Comment (seyedsharifi12) [inline exercises/C-comparison-operators/exercise.js:18]: In here I think you need to use the strict equality comparison between personA and personB.

Comment (seyedsharifi12) [reply exercises/C-comparison-operators/exercise.js:18]: other than that your code looks good.

File: exercises/D-logical-operators/exercise.js [hunk 3]
```diff
@@ -11,14 +11,14 @@ var cssLevel = 4;
...
+var htmlLevelAbove5 = htmlLevel > 5 ;
+var cssLevelAbove5 =cssLevel > 5;
 
 // Finish the next two statement
 // Use the previous variables and logical operators
 // Do not "hardcode" the answers
-var cssAndHtmlAbove5;
-var cssOrHtmlAbove5;
+var cssAndHtmlAbove5 = htmlLevel > 5 && cssLevel > 5;
+var cssOrHtmlAbove5 = htmlLevel  > 5 && htmlLevel > 5;
```
Comment (seyedsharifi12) [inline exercises/D-logical-operators/exercise.js:22]: Just a little heads up here Bimbola, in the comments it says 'Do not hardcode'. as you have already assigned the 2 variables htmlLevelAbove5 and cssLevelAbove5, you can use those variables to declare the last 2 variables. I did the same thing and then realised if the value of htmlLevel or cssLevel changed then I would have to change the rest of the code as well.

**Comment by Abed-Aghbar** [comment]
Hi Bimbola, 
Nice work, and well done👍,  your code remember me to put = in my code to make it greater than or equal.(:

**Review by seyedsharifi12 — COMMENTED** [review]
Well done Bimbola. As far as I can tell your code is neat and easy to understand. You have used descriptive variable names whenever possible. Some of the if statements could be refactored and made shorter using ternary operators but since bit is if else coursework, you have done really well.

File: exercises/I-array-properties/exercise.js [hunk 4]
```diff
@@ -6,9 +6,14 @@
 */
 
 function isEmpty(arr) {
-  return; // complete this statement
+  if(arr === 0){
+  return true; // complete this statement
+}else {
+return false;
+ }
```
Comment (nax2uk) [inline exercises/I-array-properties/exercise.js:13]: even better if 
```suggestion
return (arr.length === 0)
}
```

File: exercises/J-array-get-set/exercise.js [hunk 5]
```diff
@@ -4,12 +4,12 @@
...
 
-function first(arr) {
-  return; // complete this statement
+function first(arr) { 
+  return arr[0]; // complete this statement
 }
 
 function last(arr) {
-  return; // complete this statement
+  return arr[3]; // complete this statement
```
Comment (nax2uk) [inline exercises/J-array-get-set/exercise.js:12]:
```suggestion
  return arr[arr.length - 1]; // complete this statement
```

File: exercises/J-array-get-set/exercises2.js [hunk 6]
```diff
@@ -6,8 +6,11 @@
   - change the first value in the array to the number 1
 */
 
-var numbers = [1, 2, 3]; // Don't change this array literal declaration
-
+let numbers = [1, 2, 3]; // Don't change this array literal declaration
+number [0];
+number [1];
+number [2];
+number [3];
```
Comment (nax2uk) [inline exercises/J-array-get-set/exercises2.js:13]:
```suggestion
numbers.push(4);
```

**Review by nax2uk — COMMENTED** [review]
Hi @blasisi,

Please see feedback below. Good job!! 💪🏽

**Pull request closed without merging**